bundles/org.openhab.binding.ecovacs/src/main/java/org/openhab/binding/ecovacs/internal/api/ClientKeys.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ecovacs.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * @author Johannes Ptaszyk - Initial contribution
+ */
+@NonNullByDefault
+public class ClientKeys {
+
+    private ClientKeys() {
+        // Prevent instantiation
+    }
+
+    // TODO Find out where to get these keys and secret from to provide further information for OpenHab users.
```

Comment:
Well, without client ID/secret communication is impossible :shrug: They're extracted from Ecovacs' Android app, and - unlike services like Netatmo - there's no portal for users to register an application for the API.
I've found at least two predecents in the openhab-addons repo for hardcoded client ID/secret (the evohome and myq bindings).

I think @JohannesPtaszyk originally intended to make those configurable in case Ecovacs at some point blocks those specific credentials, but given the hard way of obtaining them, I'm not sure whether that's useful?

I'm personally inclined to remove this class and hardcode them in `EcovacsApiConfiguration`.

Reply:
I know they are needed, so it's kind of a hard question. I have just been in contact with Velux about the same issue - to get their opinion, from a legal perspective, how they will would react if a public integration would be made based on reverse engineering of their app and using the key obtained from the app. They highly recommended not doing that, but didn't directly threaten me with legal consequences, so I have sent some additional questions about the matter.

If you want the keys to be configurable it would make sense to do what you suggested, and otherwise I guess they could be moved to `EcovacsBindingConstants`. If they would change the keys, it would probably be to fight integrations, in which case they will probably also try to obscure other things at the same time.

Reply:
Moved them to `EcovacsBindingConstants`. Other integration libraries [are doing the same thing](https://github.com/mrbungle64/ecovacs-deebot.js/blob/master/library/constants.js#L23-L42) ... I doubt Ecovacs really cares, TBH.